language/src/name_resolution.rs

```diff
@@ -146,8 +170,11 @@ fn resolve_expression(
                                 (Some((new_pat, payload.1.clone())), new_name_context)
                             }
                         };
-                        let name_context = new_name_context.union(name_context.clone());
-                        let new_arm = resolve_expression(sess, arm, &name_context, top_level_ctx)?;
+                        let mut updated_name_context = name_context.clone();
+                        for (k, v) in new_name_context.into_iter() {
```

Comment:
Is there an issue with union in im 15.1.0?

Reply:
yep, see the link in the PR description (https://github.com/bodil/im-rs/issues/202)

Reply:
@cmester0 any thoughts on this?
This PR should at least unblock the update, which we should do due to the security issue.
But I think we should revisit the way the hashmaps work later. I feel like we should probably move away from im given this issue.

Reply:
It seems like a strange implementation of union, but I think the pull request is fine. I agree that we should probably find something that has the "correct" union implemented.

Reply:
@cmester0 can you r+ so I can merge? 🙂

Reply:
lgtm